packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.15;
+
+import 'forge-std/Test.sol';
+import { NounsDAOLogicV2 } from '../../contracts/governance/NounsDAOLogicV2.sol';
+import { NounsDAOProxyV2 } from '../../contracts/governance/NounsDAOProxyV2.sol';
+import { NounsDAOStorageV2 } from '../../contracts/governance/NounsDAOInterfaces.sol';
+
+contract NounsDAOLogicV2Test is Test {
+    NounsDAOLogicV2 daoLogic;
+    NounsDAOLogicV2 daoProxy;
+    address timelock = address(0x1);
+    address nouns = address(0x2);
+    address vetoer = address(0x3);
+    address admin = address(0x4);
+    uint256 votingPeriod = 6000;
+    uint256 votingDelay = 1;
+    uint256 proposalThresholdBPS = 200;
+
+    function setUp() public {
+        daoLogic = new NounsDAOLogicV2();
+
+        daoProxy = NounsDAOLogicV2(
+            payable(
+                new NounsDAOProxyV2(
+                    timelock,
+                    nouns,
+                    vetoer,
+                    admin,
+                    address(daoLogic),
+                    votingPeriod,
+                    votingDelay,
+                    proposalThresholdBPS,
+                    NounsDAOStorageV2.DynamicQuorumParams({
+                        minQuorumVotesBPS: 200,
+                        maxQuorumVotesBPS: 2000,
+                        quorumCoefficient: 10000
+                    })
+                )
+            )
+        );
+    }
+
+    function test_updateVetoer_failsIfNotCurrentVetoer() public {
+        vm.expectRevert(NounsDAOLogicV2.VetoerOnly.selector);
+        daoProxy._setPendingVetoer(address(0x1234));
+    }
+
+    event NewPendingVetoer(address oldPendingVetoer, address newPendingVetoer);
```

Comment:
nit: move events to the top of the contract

Reply:
done